src/elements/e-video.vue

```diff
@@ -0,0 +1,267 @@
+        type: Boolean,
+        default: false,
+      },
+
+      /**
+       * Accepts a valid url from a youtube or vimeo video.
+       */
+      videoUrl: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Accepts an ID from a youtube or vimeo video.
+       */
+      videoId: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Allows to set the video player width.
+       * width can either be responsive or hard coded.
+       * youtube standard -> width="560".
+       */
+      width: {
+        type: String,
+        default: null,
+      },
+
+      /**
+       * Allows to set the video player height.
+       * height can either be responsive or hard coded.
+       * youtube standard -> height="315".
+       */
+      height: {
+        type: String,
+        default: null,
+      },
+    },
+
+    // emits: {},
+    // setup(): Setup {},
+
+    computed: {
+      mapSrc(): string | undefined {
+        const playerBaseUrl = PlayerUrlSource[this.source];
+
+        if (!this.videoId && !this.videoUrl) {
+          // eslint-disable-next-line no-console
+          console.error("Neither 'videoId' or 'videoUrl' were defined", this.$el);
+
+          return undefined;
+        }
+
+        if (this.videoUrl && !this.videoId) {
+          if (!this.validateUrl()) {
+            return undefined;
+          }
+
+          let id: string | null;
+
+          switch (this.source) {
+            case VideoSource.Youtube:
+              id = new URLSearchParams(new URL(this.videoUrl).search).get('v');
+
+              return this.replaceIdInUrl(playerBaseUrl, id);
+
+            case VideoSource.Vimeo:
+              id = new URL(this.videoUrl).pathname.slice(1);
+
+              return this.replaceIdInUrl(playerBaseUrl, id);
+
+            default:
+              return undefined;
+          }
+        }
+
+        /**
+         * The videoId can consist of lowercase/uppercase letters, numbers, dashes and underscore
+         */
+        if ((/[a-zA-Z0-9-_]/).test(this.videoId) === false) {
+          // eslint-disable-next-line no-console
+          console.error('video-id can not be part of a URL');
+
+          return undefined;
+        }
+
+        return this.replaceIdInUrl(playerBaseUrl, this.videoId);
+      },
+    },
+
+    // computed: {},
+    // watch: {},
+
+    // beforeCreate() {},
+    created() {
+      if (this.width || this.height) {
+        if (!this.height && !this.width) {
+          // eslint-disable-next-line no-console
+          console.error('Both, width and height need to be specified');
+        }
+      }
+    },
+    // beforeMount() {},
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeUnmount() {},
+    // unmounted() {},
+
+    methods: {
+      /**
+       * Helper function to validate the videoUrl property.
+       */
+      validateUrl(): boolean {
```

Comment:
I'd use a computed for this (isValidUrl). Then it's always up to date and you can use it anywhere without the need of running the method again. Computed are cached internally as long as no data changes.